### ISSUE TITLE
Boilers hard-cap steam output at 100x the base rate

### ISSUE THREAD

**Issue by SuperKael** [ChiefArug/ThermalSysteams#28]
Steam boilers cannot be augmented to produce more than 100x the steam production of the base rate. This is because in `BoilerBlockEntityBase.finalizeAttributes`, `generationModifier` is clamped between `Constants.AUG_SCALE_MIN` and `Constants.AUG_SCALE_MAX`. From the name 'AUG_SCALE_MAX' you would think that this constant represents the global maximum for scale factor from augments, but that is not how this constant is actually used in standard dynamos. Instead, this constant is _only_ used to cap fuel efficiency - that is, augments cannot raise fuel efficiency past 100x the base value. Production rate, on the other hand, is not capped whatsoever. This can even be seen in this mod's own Steam Dynamo, as it happily accepts augments pushing its production rate beyond 100x the base value. It is just boilers that have their generation modifier hard-capped by `AUG_SCALE_MAX`.

I noticed this in the Star Technology modpack which adds, among other things, a 256x scale factor augment in order to allow dynamos to keep up with GregTech's crazy energy scaling. Noticing that the steam boilers were hard-capped below the expected value, I investigated and determined what was going on.

Also, I understand if you don't want to change this, but boilers also only output 1000 mb/t of steam no matter what their production rate is. External fluid extraction is required to actually pull the steam out if the production rate exceeds that. That is actually much easier to hit than the 100x scale factor cap, but is a bit less of a problem if a mod capable of extracting the steam at a sufficient rate is available.

**Comment by SuperKael** [comment]
For reference, here is the `finalizeAttributes` method from Thermal Core's `DynamoBlockEntity`:
```
    @Override
    protected void finalizeAttributes(Map<Enchantment, Integer> enchantmentMap) {

        creativeEnergy = false;

        super.finalizeAttributes(enchantmentMap);
        float baseMod = getAttributeModWithDefault(augmentNBT, TAG_AUGMENT_BASE_MOD, 1.0F);
        float processMod = getAttributeModWithDefault(augmentNBT, TAG_AUGMENT_DYNAMO_POWER, 1.0F);
        float totalMod = baseMod * processMod;

        energyStorage.applyModifiers(totalMod, totalMod);

        baseProcessTick = Math.round(getBaseProcessTick() * totalMod);
        energyMod = MathHelper.clamp(energyMod, AUG_SCALE_MIN, AUG_SCALE_MAX);

        processTick = baseProcessTick;
        minProcessTick = throttleFeature ? 0 : baseProcessTick / 10;
    }
```

**Comment by ChiefArug** [comment]
Thanks for pointing out that inconsistency, I'll remove the clamp line and that should fix it.

As for
> boilers also only output 1000 mb/t of steam no matter what their production rate is

This is intentional as it is what base Thermal does as well. Use external means of extraction if that base rate is insufficient or you want to extract it as a Mekanism gas. If a modpack adds a way to make it produce steam that stupidly fast they should make sure you can extract it that stupidly fast too.

**Comment by ChiefArug** [comment]
Fixed in fa6458f, and released on CurseForge https://www.curseforge.com/minecraft/mc-mods/systeams/files/6003506
